internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go

```diff
@@ -238,6 +306,35 @@ func resourceVirtualDesktopHostPoolUpdate(d *pluginsdk.ResourceData, meta interf
 		if d.HasChange("validate_environment") {
 			payload.Properties.ValidationEnvironment = utils.Bool(d.Get("validate_environment").(bool))
 		}
+
+		if d.HasChanges("scheduled_agent_updates_enabled", "scheduled_agent_updates_timezone", "scheduled_agent_updates_use_session_host_timezone", "agent_updates_schedule") {
+			payload.Properties.AgentUpdate = &hostpool.AgentUpdatePatchProperties{}
+
+			if d.HasChange("scheduled_agent_updates_enabled") {
+				updatesEnabled := false
+				updatesEnabled = d.Get("scheduled_agent_updates_enabled").(bool)
+				updatesScheduled := hostpool.SessionHostComponentUpdateTypeScheduled
+				updatesDefault := hostpool.SessionHostComponentUpdateTypeDefault
+				if updatesEnabled {
+					payload.Properties.AgentUpdate.Type = &updatesScheduled
+				} else {
+					payload.Properties.AgentUpdate.Type = &updatesDefault
+					payload.Properties.AgentUpdate.MaintenanceWindows = &[]hostpool.MaintenanceWindowPatchProperties{}
+				}
+			}
```

Comment:
could we do inlight assingments
```suggestion

				updatesScheduled := hostpool.SessionHostComponentUpdateTypeScheduled
				updatesDefault := hostpool.SessionHostComponentUpdateTypeDefault
				if d.Get("scheduled_agent_updates_enabled").(bool) {
					payload.Properties.AgentUpdate.Type = pointer.FromString(hostpool.SessionHostComponentUpdateTypeScheduled)
				} else {
					payload.Properties.AgentUpdate.Type = pointer.fromString()
					payload.Properties.AgentUpdate.MaintenanceWindows = &[]hostpool.MaintenanceWindowPatchProperties{}
				}
			}
```